src/contexts/DataContext.js

```diff
@@ -1,31 +1,22 @@
 import { createContext, useState } from 'react';
+import { DateTime } from 'luxon';
 
 export const DataContext = createContext();
 
 const DataContextProvider = (props) => {
 
-  const newDate = new Date();
-  const d = newDate.getDate();
-  const dd = (d < 10 ? ("0" + d) : d);
-  const m = (newDate.getMonth() + 1);
-  const mm = (m < 10 ? ("0" + m) : 0);
-  const yyyy = newDate.getFullYear();
+  const date = DateTime.local(); 
 
-  const [weatherValues, setWeatherValues] = useState([
-    {date: `${yyyy}-${mm}-${dd}`, Precipitation: 0, WindSpeed: 0},
-    {date: `${yyyy}-${mm}-${dd + 1}`, Precipitation: 0, WindSpeed: 0},
-    {date: `${yyyy}-${mm}-${dd + 2}`, Precipitation: 0, WindSpeed: 0},
-    {date: `${yyyy}-${mm}-${dd + 3}`, Precipitation: 0, WindSpeed: 0},
-    {date: `${yyyy}-${mm}-${dd + 4}`, Precipitation: 0, WindSpeed: 0},
-    {date: `${yyyy}-${mm}-${dd + 5}`, Precipitation: 0, WindSpeed: 0},
-    {date: `${yyyy}-${mm}-${dd + 6}`, Precipitation: 0, WindSpeed: 0}
-  ]);
+  const [weatherValues, setWeatherValues] = useState(
+    Array(7).fill(date).map((date, index) => {
+      var nextDate = date.plus({days: index})
+      return nextDate;
+    }).map((date) => ({date: date.toFormat('yyyy-MM-dd'), Precipitation: 0, WindSpeed: 0}))
+  );
 
   const submitWeatherValues = (weekday, precip, wind) => {
-    setWeatherValues((prevWeatherValues) =>
-      prevWeatherValues.map((weatherValue) => 
-        weatherValue.date === weekday ? {date: weekday, Precipitation: precip, WindSpeed: wind} : weatherValue
-      ));
+    setWeatherValues([...weatherValues, {date: weekday, Precipitation: precip, WindSpeed: wind}]
```

Comment:
it would be nice if we could rename these items as date, precip, and wind, like you've done above.  That way we could do this here: `[...weatherValues, { date, precip, wind }]`

Also this isn't correct as we could end up with multiples of the same object (multiples of the same date).  We should use a `Map` here  instead to prevent this.